R-notebooks/covidhub_evaluation.Rmd

```diff
@@ -49,57 +49,97 @@ forecasters = c("CMU-TimeSeries",
 forecast_dates = vector("list", length = length(forecasters))
 for (i in 1:length(forecasters)) {
   forecast_dates[[i]] = tryCatch({
-    get_forecast_dates(forecasters[i])
+    as.Date(get_forecast_dates(forecasters[i]))
   },
   error = function(e) cat(sprintf("%i. %s\n", i, e$message))
   )
 }
 ```
 
+Load data from previous run so we don't have to reingest / process it. This
+data could end up out of date if a forecast is retrospectively updated, but in
+that case it's no longer a true prediction. We can always restart from scratch
+by deleting covidhub_evaluation.rda.
+```{r}
+load(file = "covidhub_evaluation.rda")
+if (exists("predictions_cards")){
+  seen_dates = list()
+  for (i in 1:length(predictions_cards)){
+    forecaster_predictions_cards = predictions_cards[[i]]
+    pred_dates = lapply(forecaster_predictions_cards, 
+                        FUN = function(x) attributes(x)$forecast_date)
+    seen_dates[[i]] = unique(do.call("c", pred_dates))
+    name_of_forecaster = attributes(
+                          forecaster_predictions_cards[[1]])$name_of_forecaster
+    names(seen_dates)[[i]] = name_of_forecaster
+  }
+} else {
+  for (i in 1:length(forecasters)){
+    seen_dates[[i]] = list()
+  }
+  names(seen_dates) = forecasters
+}
+```
+
 Now figure out "comparable" forecast dates: making a forecast on a Sunday or a 
 Monday of the same epiweek should be comparable. **TODO**: we should switch over 
 to using the Zoltar API, and soon it should have an "as of" parameter, so then 
 we shouldn't need to do this. 
 
-```{r, eval = FALSE}
-forecast_dates_comparable = vector("list", length = length(forecasters)) 
+```{r}
+
 forecast_dates_cmu = forecast_dates[[1]]
+new_dates = list()
```

Comment:
Can you leave a comment explaining exactly what is "new" about the `new_dates`?  They are dates we don't have a comparable prediction for?

Reply:
Dates that we don't have data saved for. They are new in the sense that we will find any new forecasts that were made since the last time we ran the notebook. Technically, it includes old dates as well if they were non-comparable, because we didn't save that data. But, we filter those out when we find comparable dates.

I can add a comment.

Reply:
Please do